homeassistant/components/ekeybionyx/config_flow.py

```diff
@@ -0,0 +1,205 @@
+            ]
+            data_schema = {vol.Required("system"): SelectSelector({"options": options})}
+            return self.async_show_form(
+                step_id="choose_system", data_schema=vol.Schema(data_schema)
+            )
+        self._data["system"] = [
+            s for s in self._data["systems"] if s.system_id == user_input["system"]
+        ][0]
+        return await self.async_step_webhooks(user_input=None)
+
+    async def async_step_webhooks(
+        self, user_input: dict[str, Any] | None
+    ) -> ConfigFlowResult:
+        """Dialog to setup webhooks."""
+        system = self._data["system"]
+        await self.async_set_unique_id(system.system_id)
+        self._abort_if_unique_id_configured()
+
+        if (
+            system.function_webhook_quotas["free"] == 0
+            and system.function_webhook_quotas["used"] == 0
+        ):
+            return self.async_abort(reason="no_available_webhooks")
+
+        if system.function_webhook_quotas["used"] > 0:
+            return await self.async_step_delete_webhooks()
+
+        errors: dict[str, str] | None = None
+        if user_input is not None:
+            errors = {}
+            for webhooks in user_input.items():
+                if webhooks[0] != CONF_URL and not re.match(
+                    VALID_NAME_PATTERN, webhooks[1]
+                ):
+                    errors.update({webhooks[0]: "invalid_name"})
+            try:
+                cv.url(user_input[CONF_URL])
+            except vol.Invalid:
+                errors[CONF_URL] = "invalid_url"
+            if len([x for x in user_input.items() if x[0] != CONF_URL]) == 0:
+                errors["base"] = "no_webhooks_provided"
+
+            if not errors:
+                webhook_data = [
+                    {"webhook_id": webhook_generate_id(), "name": webhooks[1]}
+                    for webhooks in user_input.items()
+                    if webhooks[0] != CONF_URL
+                ]
+                for webhook in webhook_data:
+                    webhook.update(auth=secrets.token_hex(32))
+                    wh_def: ekey_bionyxpy.WebhookData = {
+                        "integrationName": "Home Assistant",
+                        "functionName": webhook["name"],
+                        "locationName": "Home Assistant",
+                        "definition": {
+                            "url": user_input[CONF_URL]
+                            + webhook_generate_path(webhook["webhook_id"]),
+                            "authentication": {"apiAuthenticationType": "None"},
+                            "securityLevel": "AllowHttp",
+                            "method": "Post",
+                            "body": {
+                                "contentType": "application/json",
+                                "content": json.dumps({"auth": webhook["auth"]}),
+                            },
+                        },
+                    }
```

Comment:
Okay so while I agree that the definition of the webhook is a Home Assistant concern, it feels weird to have the structure as a dict in here. As in, it feels weird that the dictation of the shape of the dict is by HA, while that would be a dict concern. So maybe it makes sense to have this in the library with objects. But just something I wanted to notice and not a blocker